### PR TITLE
Migration to SQLAlchemy for DB communication

### PR DESCRIPTION
**Pull request by Ryth-cs**

### CODE REVIEW

File: src/requirements.txt [hunk 1]
```diff
@@ -1,12 +1,12 @@
 psycopg2-binary>=2.8
+sqlalchemy
 discord.py
```
Comment (Fluxticks) [inline src/requirements.txt:3]: Do we want to move this to be discord.py[voice]?

**Review by Fluxticks — APPROVED** [review]
Looks good and is working when it comes to the music cog side of things.

File: src/esportsbot/base_functions.py [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
+from esportsbot.db_gateway_v1 import DBGatewayActions
+from esportsbot.models import Voicemaster_master, Voicemaster_slave, Guild_info
 
 
 async def send_to_log_channel(self, guild_id, msg):
-    db_logging_call = db_gateway().get('guild_info', params={'guild_id': guild_id})
-    if db_logging_call and db_logging_call[0]['log_channel_id']:
-        await self.bot.get_channel(db_logging_call[0]['log_channel_id']).send(msg)
+    db_logging_call = DBGatewayActions().get(Guild_info, guild_id=guild_id)
+    if db_logging_call and db_logging_call.log_channel_id is not None:
```
Comment (Ryth-cs) [inline src/esportsbot/base_functions.py:7]: "is not None" can be removed

File: src/esportsbot/cogs/EventCategoriesCog.py [hunk 3]
```diff
@@ -74,28 +75,26 @@ async def getGuildEventSettings(self, ctx: Context, eventName: str) -> Tuple[dic
...
+        guild = DBGatewayActions().get(Guild_info, guild_id=ctx.guild.id)
+        no_shared_role = guild.shared_role_id is None
+        if no_shared_role:
             await ctx.message.reply(self.STRINGS['no_shared_role'].format(command_prefix=self.bot.command_prefix))
         else:
-            eventData = db.get("event_categories", params={"guild_id": ctx.guild.id, "event_name": eventName})[0]
+            eventData = DBGatewayActions().get(Event_categories, guild_id=ctx.guild.id, event_name=eventName)
             if not eventData:
-                if not (allEvents := db.get("event_categories", params={"guild_id": ctx.guild.id})):
+                if not (allEvent := DBGatewayActions().list(Event_categories, guild_id=ctx.guild.id)):
```
Comment (Ryth-cs) [inline src/esportsbot/cogs/EventCategoriesCog.py:85]: Wrong var name

File: src/esportsbot/cogs/EventCategoriesCog.py [hunk 4]
```diff
@@ -611,26 +595,24 @@ async def admin_cmd_unregister_event_category(self, ctx: Context, *, args: str):
...
                 else:
                     await ctx.message.reply(
-                        self.STRINGS['unrecognised_event'].format(
-                            events=", ".join(e["event_name"].title() for e in allEvents)
-                        )
+                        self.STRINGS['unrecognised_event'].format(events=", ".join(e.event_name.title() for e in allEvents))
                     )
             else:
-                db.delete("event_categories", {"guild_id": ctx.guild.id, "event_name": eventName})
+                event_category = DBGatewayActions().get(Event_categories, guild_id=ctx.guild.id, event_name=eventName)
```
Comment (Ryth-cs) [inline src/esportsbot/cogs/EventCategoriesCog.py:607]: This doesn't need fetching twice

File: src/esportsbot/cogs/LogChannelCog.py [hunk 5]
```diff
@@ -14,14 +14,12 @@ def __init__(self, bot):
...
-        log_channel_exists = db_gateway().get('guild_info', params={'guild_id': ctx.author.guild.id})
-        if bool(log_channel_exists):
-            if log_channel_exists[0]['log_channel_id'] != cleaned_channel_id:
-                db_gateway().update(
-                    'guild_info',
-                    set_params={'log_channel_id': cleaned_channel_id},
-                    where_params={'guild_id': ctx.author.guild.id}
-                )
+        guild = DBGatewayActions().get(Guild_info, guild_id=ctx.author.guild.id)
+        log_channel_exists = guild.log_channel_id is not None
```
Comment (Ryth-cs) [inline src/esportsbot/cogs/LogChannelCog.py:18]: Need to check guild returns something

File: src/esportsbot/cogs/LogChannelCog.py [hunk 6]
```diff
@@ -42,24 +41,23 @@ async def setlogchannel(self, ctx, given_channel_id=None):
     @commands.command()
     @commands.has_permissions(administrator=True)
     async def getlogchannel(self, ctx):
-        log_channel_exists = db_gateway().get('guild_info', params={'guild_id': ctx.author.guild.id})
+        guild = DBGatewayActions().get(Guild_info, guild_id=ctx.author.guild.id)
+        log_channel_exists = guild.log_channel_id is not None
```
Comment (Ryth-cs) [inline src/esportsbot/cogs/LogChannelCog.py:45]: What if guild is None

File: src/esportsbot/lib/client.py [hunk 7]
```diff
@@ -160,38 +160,38 @@ async def init(self):
...
+                for roleData in DBGatewayActions().list(Pingable_roles, guild_id=guildData.guild_id):
+                    role = guild.get_role(roleData.role_id)
                     if role is None:
-                        print("[Esportsbot.init] Unknown pingable role id in pingable_roles table. Removing from the table: role #" \
-                                + str(roleData["role_id"]) + " in guild #" + str(guildData["guild_id"]))
-                        db.delete("pingable_roles", {"role_id": roleData["role_id"]})
+                        print(
+                            f"[Esportsbot.init] Unknown pingable role id in pingable_roles table. Removing from the table: role #{roleData.role_id} in guild #{guildData.guild_id}"
+                        )
+                        pingable_role = DBGatewayActions().get(Pingable_roles, role_id=roleData.role_id)
```
Comment (Ryth-cs) [inline src/esportsbot/lib/client.py:189]: Redundant, can delete roleData as in interator

**Pull request merged**